cookbooks/aws-parallelcluster-platform/resources/fetch_config.rb

```diff
@@ -33,6 +34,10 @@
       fetch_instance_type_data unless ::FileUtils.identical?(node['cluster']['previous_cluster_config_path'], node['cluster']['cluster_config_path'])
       Chef::Log.info("Backing up old shared storages data from (#{node['cluster']['shared_storages_mapping_path']}) to (#{node['cluster']['previous_shared_storages_mapping_path']})")
       ::FileUtils.cp_r(node['cluster']['shared_storages_mapping_path'], node['cluster']['previous_shared_storages_mapping_path'], remove_destination: true)
+
+      Chef::Log.info("Updating #{login_node_keys_sync_file} during #{node['cluster']['node_type']} update")
+      write_sync_file(login_node_keys_sync_file)
```

Comment:
This fix would address the issue, but we should  do something easier: in this sync file we should not even check the cluster config version at all because it's enough for it to be there to signal that the head node already created the keys. In the offending PRs we included the check of the cluster config version to replicate an approach that was equivalent to the sync file used by the head node to signal changes in the config file. In this case though, the check of the cluster config version is redudant and can be avoided.